test experiment script

gcramer23 opened the pull request:
1. adds test_scripts.py that will run added scripts and verify that there are no errors
2. adds local ddp_nccl_allreduce experiment script

test with command `pytest test_scripts.py`

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57925 test experiment script**

Differential Revision: [D28382452](https://our.internmc.facebook.com/intern/diff/D28382452)

gcramer23 committed:
test experiment script

[ghstack-poisoned]

gcramer23 committed:
Update on "test experiment script"




[ghstack-poisoned]

gcramer23 committed:
Update on "test experiment script"

1. adds test_scripts.py that will run added scripts and verify that there are no errors
2. adds local, slurm interactive, and slurm batch scripts for ddp_nccl_allreduce experiment

test with command `pytest test_scripts.py`




[ghstack-poisoned]

gcramer23 commented on benchmarks/distributed/rpc/parameter_server/test_scripts.py:
```diff
@@ -0,0 +1,23 @@
+    assert not error
+
+
+def test_ddp_nccl_allreduce():
```
it is not a unit test

mrshenli commented on benchmarks/distributed/rpc/parameter_server/experiment_scripts/ddp_nccl_allreduce_sbatch.sh:
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
shall we remove slurm related scripts? If you would like to keep this somewhere, is it OK to create a repo in your own github space? It's a little weird to have SLURM commands in PT source code

gcramer23 replied:
okay.

gcramer23 committed:
Update on "test experiment script"

1. adds test_scripts.py that will run added scripts and verify that there are no errors
2. adds local, slurm interactive, and slurm batch scripts for ddp_nccl_allreduce experiment

test with command `pytest test_scripts.py`




[ghstack-poisoned]

mrshenli commented on benchmarks/distributed/rpc/parameter_server/test_scripts.py:
```diff
@@ -0,0 +1,23 @@
+import subprocess
```
curious, do we expect this .py file or the above .sh file to be the entrance script? I'd assume it's the .sh file, as running `python test_scripts.py` won't do anything IIUC. In this case, what's the purpose of having this script?

gcramer23 replied:
The entrance will be .py file for checking errors using pytest. And .sh file for running the benchmark using python. 

A user can use the script to check subsets of the experiment scripts added to the test script to see if the subset is passing. I think it will make checking the scripts easier for larger subset sizes.

gcramer23 commented:
@gcramer23 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28382452).
